va/va.go

```diff
@@ -757,6 +800,51 @@ func (va *ValidationAuthorityImpl) validateChallenge(ctx context.Context, identi
 	return nil, probs.Malformed(fmt.Sprintf("invalid challenge type %s", challenge.Type))
 }
 
+func (va *ValidationAuthorityImpl) performRemoteValidation(ctx context.Context, domain string, challenge core.Challenge, authz core.Authorization, result chan *probs.ProblemDetails) {
+	s := va.clk.Now()
+	errors := make(chan error, len(va.remoteVAs))
+	wg := new(sync.WaitGroup)
+	for _, remoteVA := range va.remoteVAs {
+		wg.Add(1)
+		go func(rva RemoteVA) {
+			defer wg.Done()
+			_, err := rva.PerformValidation(ctx, domain, challenge, authz)
+			if err != nil {
+				// returned error can be a nil *probs.ProblemDetails which breaks the
+				// err != nil check so do a slightly more complicated unwrap check to
+				// make sure we don't choke on that.
+				if p, ok := err.(*probs.ProblemDetails); !ok || p != nil {
+					va.log.Info(fmt.Sprintf("Remote VA %q.PerformValidation failed: %s", rva.Address, err))
+					errors <- err
+				}
+			}
+		}(remoteVA)
+	}
+	wg.Wait()
```

Comment:
@rolandshoemaker @jsha Have we considered whether it would be better to allow the design to return from `performRemoteValidation` when `len(va.remoteVAs) - maxRemoteFailures`  have returned positive answers? E.g. in the case where we had 5 distributed VAs and tolerated 2 remote failures, we would return after 3 successful validations. Josh pointed out that otherwise if one remote VA was misbehaving (maybe upstream network issues?) it would increase every validation attempt's overall time to the timeout of the one misbehaving remote VA.